### PR TITLE
[red-knot] extract helper functions in inference tests

### PR DESCRIPTION
**Pull request by carljm**
There's a lot of repeat boilerplate in the type inference tests; this cuts it down a lot.

### CODE REVIEW

File: crates/red_knot/src/types/infer.rs [hunk 1]
```diff
@@ -237,67 +238,74 @@ mod tests {
...
+    fn assert_public_type(
+        case: &TestCase,
+        modname: &str,
+        varname: &str,
+        tyname: &str,
+    ) -> anyhow::Result<()> {
+        let ty = get_public_type(case, modname, varname)?;
 
+        let jar = HasJar::<SemanticJar>::jar(&case.db)?;
+        assert_eq!(format!("{}", ty.display(&jar.type_store)), tyname);
```
Comment (AlexWaygood) [inline crates/red_knot/src/types/infer.rs:276]: It feels kinda unfortunate that we have to convert the type to a string in order to assert that it is what we expect. But I agree that this is for the best, as it's going to allow us to write much more readable tests.

**Review by AlexWaygood — APPROVED** [review]
This is great!

File: crates/red_knot/src/types/infer.rs [hunk 2]
```diff
@@ -237,67 +238,74 @@ mod tests {
...
-        let e_sym = a_syms
-            .root_symbol_id_by_name("E")
-            .expect("E symbol should be found");
+        let syms = symbol_table(db, file)?;
+        let sym = syms
+            .root_symbol_id_by_name(varname)
+            .expect("symbol should be found");
 
-        let ty = infer_symbol_public_type(
+        Ok(infer_symbol_public_type(
```
Comment (MichaReiser) [inline crates/red_knot/src/types/infer.rs:258]: Could you use `resolve_global_module` instead`?

**Review by MichaReiser — APPROVED** [review]

File: crates/red_knot/src/types/infer.rs (hunk 1 above)
Comment (carljm) [reply crates/red_knot/src/types/infer.rs:276]: On the other hand, we may eventually want to write tests based on `reveal_type` or `assert_type` anyway, which also use the textual representation of a type.

File: crates/red_knot/src/types/infer.rs (hunk 2 above)
Comment (carljm) [reply crates/red_knot/src/types/infer.rs:258]: Good point! Done. I think this code predated `resolve_global_symbol`, and I just rearranged the existing code here without thinking about it too much.

**Pull request merged**